### PR TITLE
Adds a multi-dimensional bomb payload to the black market.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
This PR adds a !!!FUN!!! bomb payload to the blackmarket, which, upon detonation, transmutates all terrain in a range like the dimensional anomaly would. You can also select the dimensional theme to use by using it in your hand.

I believe however, this thing should cost a fuckton to get and only show up occasionally. 8k to 10k was my original idea, but I think that's perhaps not high enough given its flagrant potential. Perhaps I should also make it so the more dangerous themes yield a shorter range than others. Suggestions are welcome.

Screenshot of what happens when you don't set the theme:
chaotic_mess

This PR also turns dimension themes into singletons so we access them more easily. Nothing to write home about.

## Why It's Good For The Game
The black market could always use some extra thingy or two anyway, and this thing could either be a source of emergent gameplay, or a recipe for a disaster. Perhaps second to the Big Slappy for how funny it could be.

## Changelog

:cl:
add: Added a multi-dimensional bomb payload to the black market. It's very expensive.
/:cl:

### CODE REVIEW

**Review by ZephyrTFA — APPROVED** [review]
yes

**Comment by HoodMilk** [comment]
Brb turning the armory into pizza and then eating the walls to get in.

File: code/controllers/subsystem/materials.dm [hunk 1]
```diff
@@ -47,6 +50,8 @@ SUBSYSTEM_DEF(materials)
 			continue // Do not initialize at mapload
 		InitializeMaterial(list(mat_type))
 
+	dimensional_themes = init_subtypes_w_path_keys(/datum/dimension_theme)
```
Comment (MrMelbert) [inline code/controllers/subsystem/materials.dm:53]: Small thing about making them singletons, while most of the themes don't hold state, `/datum/dimension_theme/fancy` does. It randomly selects 1 carpet type to use as a floor.

This means the fancy theme will always pick the same carpet on a per-round basis. If you're ok with that then it's whatever, just worth pointing out.

**Comment by SomeoneYouProbablyKnow** [comment]
This SHOULD be Avaliable to clown operatives, mostly because the screenshot hurts me on a spiritual level and that's what clowns are all about.

File: code/controllers/subsystem/materials.dm (hunk 1 above)
Comment (Ghommie) [reply code/controllers/subsystem/materials.dm:53]: Oh, thank you. I'll make sure to fix that.

**Comment by Admiral-Gigaton** [comment]
What is the range

Can this be detonated via something like c4 attached to it, like a normal payload

Can you add a super lube or bananium theme for maximum chaos

Full on cult theme (walls, doors, floors, but without the stun from nar nar doors)

Rat'var bronze theme

A debug version for admin fuckery, so threat ones (uranium fever) are as big as a normal one that isn't a threat theme?

Some sort of emagging tomfoolery for expanded access to shart-ass themes

**Comment by Ghommie** [comment]
The range is 17 right now, big enough to cover a good chunk of any department.
Normal payloads can be detonated with c4, syndie ones cannot.
This PR doesn't add any more themes, but you can always code some yourself and they'll be selectable.
There's no need for a debug version. The range can be varedited, and selecting the theme can be done by anyone with a good ol' pair of hands.
There's no plan for a EMAG-only theme for the time being.

**Comment by carlarctg** [comment]
This is unbelievably god-awful and objectively a bad idea and I love it.

**Comment by KingkumaArt** [comment]
> This is unbelievably god-awful and objectively a bad idea and I love it.

God I love ss13

**Comment by Ghommie** [comment]
> it seems to be broken

Fixed in #81721.

**Pull request merged**